### ISSUE TITLE
Unions not working as aspected

### ISSUE THREAD

**Issue by devtim123** [Urigo/graphql-modules#972]
Hey, as i mentioned in the close ticket https://github.com/Urigo/graphql-modules/issues/619 i can not resolve union types by using a standalone project wide graphql-modules schema. I don't know if the reason could be that i using typegraphql or apollo-server.

i wrote a simple GraphQL-Apollo-Server example demonstrate a bug when using unions in combination with graphql-modules + typegraphql + apollo-server. would like to know how to solve this problem or how to debug it. GitHub-Repo: https://github.com/devtim123/graphql-modules-union-example

commands to reproduce the issue:
```
git clone git@github.com:devtim123/graphql-modules-union-example.git
cd graphql-modules-union-example
yarn install
yarn dev
```
and visit http://localhost:4000/api

start query at playground:
```
query example {
  example {
    __typename
    ... on Actor {
      name
      age
    }
    ... on Movie {
      name
      rating
    }
  }
}

```

expected result:
```
{
  "data": {
    "example": {
      "__typename": "Movie",
      "name": "Fight Club",
      "rating": 999
    }
  }
}
```

current result:
```
{
  "data": {
    "example": {
      "__typename": "Movie"
    }
  }
}
```

For this simple example, creating the schema via (see Line #15 @ src/server.ts), the resolver works as expected:

```
const schema = await buildSchema({
    resolvers: [rootModule.resolvers]
});
```

But this works not in larger GraphQL-Modules projects, where i get the following error on startup:

```
TypeError: Cannot convert undefined or null to object
```

I hope we can find a solution there...

thank you for any help, best regards, tim

**Comment by fridaystreet** [comment]
I'm also seeing this issue still happening and I'm passing my top level module schema to apollo-server. Still the __resolverType function is never called.

Any further thoughts on this?

**Comment by kamilkisiela** [comment]
We released a new major version (v1.0) of GraphQL Modules. Here's the [migration guide](https://graphql-modules.com/docs/next/recipes/migration).

The new version solves the issue, so I'm closing the thread.

Please try to upgrade. If you can't and still want to use the legacy version, please open a new issue.
